source/normalize-arguments.js

```diff
@@ -187,24 +186,14 @@ module.exports = (url, options, defaults) => {
 		delete options.timeout;
 	}
 
-	if (is.nullOrUndefined(options.hooks)) {
-		options.hooks = {};
-	}
 	if (is.object(options.hooks)) {
-		for (const hookEvent of knownHookEvents) {
-			const hooks = options.hooks[hookEvent];
-			if (is.nullOrUndefined(hooks)) {
-				options.hooks[hookEvent] = [];
-			} else if (is.array(hooks)) {
-				hooks.forEach(
-					(hook, index) => {
-						if (!is.function_(hook)) {
-							throw new TypeError(
-								`Parameter \`hooks.${hookEvent}[${index}]\` must be a function, not ${is(hook)}`
-							);
-						}
+		for (const [hookEvent, hooks] of Object.entries(options.hooks)) {
+			if (is.array(hooks)) {
```

Comment:
This fails to validate the known hooks if something other than an array is provided, which will result in an error downstream.

Reply:
Nope. https://github.com/szmarczak/got/blob/b22c3abbe8daf905aae4bed66cddfc7b9ed72170/source/normalize-arguments.js#L191-L199

Reply:
Ah, thanks - I got tripped up reviewing the diff.